test/has_properties_test.rb

```diff
@@ -124,6 +124,28 @@ def mock.render?; true; end
 
       assert_match /w:pPr b="true"/, builder.to_xml
     end
+
+    should "allow properties to be rendered as direct children" do
+      element = Class.new(OpenXml::Element) do
+        include OpenXml::HasProperties
+        tag :bodyPr
+        namespace :a
+
+        properties_are_children
```

Comment:
This API seems a little awkward — it seems weird to have to explicitly state this when defining a new element.

Is the difference that —

normally we expect it to work like this:

```
<p>
  <pPr>
    <bold/>
  </pPr>
</p>
```

but sometimes it works like this?


```
<p>
  <bold/>
</p>
```

* * * 

What if you could mix one of two modules into an element: `HasProperties` or `ContainsProperties` (or something like that) and one of those modules was implemented in terms of the other?

Reply:
This is exactly the difference. I like the idea of having two separate modules instead of a magic method. I'll have another go at it with that approach.